Compile/src/Producers/PrefiringWeightProducer.cc

```diff
@@ -1,162 +1,338 @@
+                if (fabs(eta_jet) > 3.)
+                    continue;
+                if (jetMaxMuonFraction_ > 0 && jet->muonFraction > jetMaxMuonFraction_)
+                    continue;
+                //Loop over photons to remove overlap
+                double nonprefiringprobfromoverlappingphotons = 1.;
+                bool foundOverlappingPhotons = false;
+                for (const auto& photon : thePhotons) {
+                    double pt_gam = photon->p4.Pt();
+                    double eta_gam = photon->p4.Eta();
+                    double phi_gam = photon->p4.Phi();
+                    if (pt_gam < 20.)
+                        continue;
+                    if (fabs(eta_gam) < 2.)
+                        continue;
+                    if (fabs(eta_gam) > 3.)
+                        continue;
+                    double dR2 = ROOT::Math::VectorUtil::DeltaR2(jet->p4, photon->p4);
+                    if (dR2 > 0.16)
+                        continue;
+                    double prefiringprob_gam = getPrefiringRateEcal(eta_gam, pt_gam, h_prefmap_photon_, fluct);
+                    nonprefiringprobfromoverlappingphotons *= (1. - prefiringprob_gam);
+                    foundOverlappingPhotons = true;
+                }
+                //useEMpt =true if one wants to use maps parametrized vs Jet EM pt instead of pt.
+                if (useEMpt_)
+                    pt_jet *= (jet->neutralHadronFraction + jet->chargedHadronFraction);
+                double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRateEcal(eta_jet, pt_jet, h_prefmap_jet_, fluct);
 
+                if (!foundOverlappingPhotons) {
+                    nonPrefiringProbaECAL[fluct] *= nonprefiringprobfromoverlappingjet;
+                }
+                //If overlapping photons have a non prefiring rate larger than the jet, then replace these weights by the jet one
+                else if (nonprefiringprobfromoverlappingphotons > nonprefiringprobfromoverlappingjet) {
+                    if (nonprefiringprobfromoverlappingphotons > 0.) {
+                        nonPrefiringProbaECAL[fluct] *= nonprefiringprobfromoverlappingjet / nonprefiringprobfromoverlappingphotons;
+                    } else {
+                        nonPrefiringProbaECAL[fluct] = 0.;
+                    }
+                }
+                //Last case: if overlapping photons have a non prefiring rate smaller than the jet, don't consider the jet in the event weight, and do nothing.
             }
-            double nonprefiringprobfromoverlappingjet = 1. - getPrefiringRate(eta_jet, pt_jet, h_prefmap_jet, fluct);
-            if (nonprefiringprobfromoverlappingphotons == 1.)
-            {
-                nonPrefiringProba[fluct] *= nonprefiringprobfromoverlappingjet;
+        }
+        //Now calculate prefiring weights for muons
+        if (!missingInputMuon_) {
+            for (KMuons::iterator muon = theMuons->begin(); muon != theMuons->end(); ++muon) {
```

Comment:
Is there a reason not to use auto here (as above)?
(I am just wondering :D)

Reply:
It won't compile with auto for whatever reason. So I did it manually.

Reply:
#C++-Things :D